src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -681,6 +682,17 @@ private BoundExpression BoxReceiver(BoundExpression rewrittenReceiver, NamedType
             switch (expression.Kind)
             {
                 case BoundKind.ThisReference:
+                    if (userCodeCanExecuteBetweenReads)
+                    {
+                        // if user code could execute between reads, then the value of `this` on a struct can change
+                        var @this = (BoundThisReference)expression;
+                        return @this.Type.IsStructType();
```

Comment:
> @this.Type.IsStructType(); [](start = 30, length = 27)

Should we check that we are in a constructor of this type?

Reply:
I messed up. This change is not intended for this PR (I have a dedicated PR for the issue with `this`). I'll remove.

---
In reply to: [173937653](https://github.com/dotnet/roslyn/pull/25396#discussion_r173937653) [](ancestors = 173937653)